File: exts/jphp-gui-ext/src/main/resources/JPHP-INF/sdk/php/gui/paint/UXRadialGradient.php
<?php


namespace php\gui\paint;


/**
 * Class UXRadialGradient
 * @package php\gui\paint
 * @packages gui, javafx
 */
class UXRadialGradient extends UXPaint
{
    /**
     * @readonly
     * @var double
     */
    public $centerX;
    /**
     * @readonly
     * @var double
     */
    public $centerY;
    /**
     * @readonly
     * @var double
     */
    public $focusAngle;
    /**
     * @readonly
     * @var double
     */
    public $focusDistance;
    /**
     * @readonly
     * @var double
     */
    public $radius;

    /**
     * @readonly
     * @var UXStop[]
     */
    public $stops;
    /**
     * @readonly
     * @var boolean
     */
    public $proportional;
    /**
     * @readonly
     * @var string
     * NO_CYCLE, REFLECT, REPEAT
     */
    public $cycleMethod;


    /**
     * UXRadialGradient constructor.
     * @param float $focusAngle
     * @param float $focusDistance
     * @param float $centerX
     * @param float $centerY
     * @param float $radius
     * @param float $proportional
     * @param string $cycleMethod
     * @param array $stops
     */
    public function __construct($focusAngle, $focusDistance, $centerX, $centerY, $radius, $proportional, $cycleMethod, $stops){

    }

    /**
     * @param string $value
     * @return UXRadialGradient
     */
    public static function of($value){}
}

File: exts/jphp-gui-ext/src/main/java/org/develnext/jphp/ext/javafx/classes/paint/UXRadialGradient.java
package org.develnext.jphp.ext.javafx.classes.paint;

import javafx.scene.paint.CycleMethod;
import javafx.scene.paint.LinearGradient;
import javafx.scene.paint.RadialGradient;
import javafx.scene.paint.Stop;
import org.develnext.jphp.ext.javafx.JavaFXExtension;
import php.runtime.annotation.Reflection;
import php.runtime.env.Environment;
import php.runtime.reflection.ClassEntity;

import java.util.List;

@Reflection.Name(JavaFXExtension.NS + "paint\\UXRadialGradient")
public class UXRadialGradient extends UXPaint<RadialGradient>{

    public UXRadialGradient(Environment env, RadialGradient wrappedObject) {
        super(env, wrappedObject);
    }

    public UXRadialGradient(Environment env, ClassEntity clazz) {
        super(env, clazz);
    }

    interface WrappedInterface{
        @Reflection.Getter CycleMethod getCycleMethod();

        @Reflection.Getter double getCenterX();
        @Reflection.Getter double getCenterY();

        @Reflection.Getter double getFocusAngle();
        @Reflection.Getter double getFocusDistance();
        @Reflection.Getter double getRadius();

        @Reflection.Getter List<Stop> getStops();

        @Reflection.Getter("proportional") boolean isProportional();
    }

    @Reflection.Signature
    public void __construct(double focusAngle, double focusDistance, double centerX, double centerY, double radius, boolean proportional, CycleMethod cycleMethod, List<Stop> stops){
        __wrappedObject = new RadialGradient(focusAngle, focusDistance, centerX, centerY, radius, proportional, cycleMethod, stops);
    }

    @Reflection.Signature
    public static RadialGradient of(String value){
        try{
            return RadialGradient.valueOf(value);
        }
        catch(IllegalArgumentException e){
            return null;
        }
    }
}